ai_agents/travel_planner_agent/agent.py

```
@@ -0,0 +1,309 @@
+
+@tool
+def research_destination(query: str) -> str:
+    """Search the web for destination highlights, attractions, food, and travel tips.
+
+    Args:
+        query: Search query (e.g. 'best things to do in Kyoto Japan 2025').
+    """
+    try:
+        with DDGS() as ddgs:
+            results = list(ddgs.text(query, max_results=6))
+    except Exception as exc:
```

Ruff (0.15.13)

[warning] Do not catch blind exception: `Exception`

(BLE001)